vitals_monitor.cpp

```diff
@@ -0,0 +1,119 @@
+};
+
+// List of conversion rules
+std::vector<ConversionRule> conversionRules = {
+    {CELSIUS, FAHRENHEIT, [](double value) { return value * 9.0 / 5.0 + 32; }},
+    {FAHRENHEIT, FAHRENHEIT, [](double value) { return value; }},
+    {BPM, BPM, [](double value) { return value; }},
+    {PERCENTAGE, PERCENTAGE, [](double value) { return value; }},
+};
+
+// Critical statuses that require animation
+static const std::vector<VitalStatus> criticalStatuses = {LOW, HIGH};
+
+// Star animation function for critical alerts
+void showAlertAnimation(int durationInSeconds) {
+    for (int i = 0; i < durationInSeconds * 2; ++i) {
+        std::cout << (i % 2 == 0 ? "\r* " : "\r *") << std::flush;
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    }
+    std::cout << "\n";
+}
+
+void printVitalStatus(const VitalData& vital, const std::map<VitalStatus,
+                                              std::string>& messageMap) {
+    // Find the alert message for the current vital type and status
+    auto messageIt = messageMap.find(vital.status);
+
+    // Print the message if found
+    if (messageIt != messageMap.end()) {
+        std::cout << messageIt->second << std::endl;
+    }
+}
+
+// Function to print alert based on the vital's status using the selected message map and language
+void printAlert(const VitalData& vital, Language language) {
+    // Find the alert message for the current vital type and status in the selected language
+    printVitalStatus(vital, vitalMessages.at(language).at(vital.type));
+
+    // Check if status is critical and trigger animation
+    if (std::find(criticalStatuses.begin(), criticalStatuses.end(),
+                                            vital.status) != criticalStatuses.end()) {
+        showAlertAnimation(2);
+    }
+}
+
+// Function to convert values to the appropriate unit using vector functions
+double convertToCommonUnit(double value, VitalUnit fromUnit) {
+    // Find the matching conversion rule
+    auto it = std::find_if(conversionRules.begin(), conversionRules.end(),
+                           [fromUnit](const ConversionRule& rule) {
+                               return rule.fromUnit == fromUnit;
+                           });
+
+    // Apply the conversion function if a rule is found
+    return (it != conversionRules.end()) ? it->convertFunc(value) : value;
+}
+
+// Function to check and update the vital status based on the thresholds
+void checkVitalStatus(VitalData* vital) {
+    if (vital == nullptr) return;  // Guard clause for null pointers
+
+    double convertedValue = convertToCommonUnit(vital->value, vital->unit);
+    const VitalLimits& limits = vitalThresholds.at(vital->type);
+
+    // Array of thresholds to map ranges to status
+    std::vector<std::pair<double, VitalStatus>> thresholds = {
+        {limits.low, LOW},
+        {limits.nearLow, NEAR_LOW},
+        {limits.nearHigh, NORMAL},
+        {limits.high, NEAR_HIGH}

```

Comment:
There's no `HIGH` here, though it's present in the enum. It's also missing from the tests?
Is that implementation still pending?